File: Python/Algorithms/binary-search.py
def rec_binary_search(array, element, start, end):
    if start > end:
        return -1
    
    mid = (start + end) // 2
    if element == array[mid]:
        return mid
    
    elif element < array[mid]:
        return rec_binary_search(array, element, start, mid-1)
    
    else:
        return rec_binary_search(array, element, mid+1, end)

element = 18
array = [1, 2, 5, 7, 13, 15, 16, 18, 24, 28, 29]

print(f"Searching for {element}")
print(f"Index of {element}: {rec_binary_search(array, element, 0, len(array))}")

File: TypeScript/search-algos/binary-search.js
var binarySearch = function (nums, key) {
    var low = 0;
    var high = nums.length - 1;
    while (low <= high) {
        var mid = Math.floor((low + high) / 2);
        mid;
        if (nums[mid] === key) {
            return mid + 1;
        }
        if (key > nums[mid]) {
            low = mid + 1;
        }
        else {
            high = mid - 1;
        }
    }
    return -1;
};
var array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
var sKey = 6;
console.log("Searching for ".concat(sKey));
console.log("Index of ".concat(sKey, " : ").concat(binarySearch(array, sKey)));
